### PR TITLE
Upgrade to openfisca core 30

### PR DESCRIPTION
**Pull request by Br3nda**
## Breaking changes
 - no longer accepts MONTH period values for variables that are by DAY
 - age variables must be set for a DAY period, not MONTH or YEAR

### CODE REVIEW

File: openfisca_aotearoa/tests/social_security/child.yaml [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
-        age: 6
+        age:
+          "2018-08-01": 6
       Ringo:
-        age: 17
-        social_security__is_financially_independent: true
+        age:
+          "2018-08-01": 17
+        social_security__is_financially_independent:
+          "2018-08": true
```
Comment (dlouise64) [inline openfisca_aotearoa/tests/social_security/child.yaml]: @Br3nda does this need a day too?

Comment (Br3nda) [reply openfisca_aotearoa/tests/social_security/child.yaml]: No. That variable had a period = MONTH

**Comment by Br3nda** [comment]
@Catalyst have looked into the [SmartStart](http://github.com/catalyst/smartstart-backend/) app and they are passing age for a month. They need to update to support by `DAY` or use `date_of_birth`. They haven't released their recent source code for the last year, so they're looking to fix that too, so we can read their code and test ourselves next time.

**Comment by Br3nda** [comment]
rebased

**Comment by Br3nda** [comment]
Scheduled for go live, tomorrow, 1:30pm-ish

**Pull request merged**